docs/zkapps/advanced-snarkyjs/time-locked-accounts.mdx

```diff
@@ -0,0 +1,114 @@
+---
+title: Time-Locked Accounts
+hide_title: true
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Time-Locked Accounts
+
+Time-locking allows you to pay someone MINA or other tokens, which are subject to a vesting schedule. This means that the tokens are initially locked, and only become available for withdrawal after a certain time, or gradually according to a certain schedule.
+
+The zkApp feature that enables time-locking is the `timing` field, which is present on every account. It look like this:
+
+```ts
+type Account = {
+  // ...
+  timing: {
+    isTimed: Bool;
+    initialMinimumBalance: UInt64;
+    cliffTime: UInt32;
+    cliffAmount: UInt64;
+    vestingPeriod: UInt32;
+    vestingIncrement: UInt64;
+  };
+};
+```
+
+`isTimed` tells us whether this account is subject to any time-locking. The other fields are parameters that allow us to define a vesting schedule in a very flexible manner. We will explain them in detail below. By default, an account has `isTimed: false`; in that case, the other fields are dummy values.
+
+At a glance, this graph shows you how each of the timing parameters affect the vesting schedule:
+
+<figure>
+  <img
+    src="/img/10_time_locked_accounts.png"
+    width="95%"
+    alt="Timing parameters"
+  />
+</figure>
+
+The red cross on the left marks the point in time where the `timing` field is set and `isTimed` switches from false to true. The blue line shows how the amount of unlocked tokens increases over time, until it finally reaches its maximum value and stays flat. At this point, `isTimed` flips from `true` back to `false`, because there no longer are any tokens locked.
```

Comment:
> `false` to `true`
backticks for these

`, because there no longer are any tokens locked.` > `, because no tokens remain locked.`